### DISCUSSION TITLE
Invalid YAML cannot be fixed

### DISCUSSION THREAD

**Discussion by Tyler887** [community/community#25780] [Actions]
<p>I am trying to create an Action, but my YAML syntax is invalid:</p>
<pre><code class="lang-yaml">- name: Publish artifacts
  runs-on: ubuntu-latest

on:
 [release]
steps:
  - name: Cache the artifact
    uses: actions/cache@v2.1.7
</code></pre>
<p>I cannot resolve this. Can you help?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>The YAML syntax is invalid because the top level of your file (zero indentation) has both a list element and a mapping. The <a href="https://en.wikipedia.org/wiki/YAML">Wikipedia article on YAML</a> has a description of the syntax, I like the <a href="https://docs.ansible.com/ansible/latest/reference_appendices/YAMLSyntax.html">YAML introduction in the Ansible documentation</a>, and of course you can also check the <a href="https://yaml.org/spec/1.2.2/#chapter-2-language-overview">YAML specification</a> (but that might not be the most beginner friendly).</p>
<p>After fixing that, your file lacks a <code>jobs</code> element. I assume the first two lines and <code>steps</code> element should be combined into one job. Just a cache step isn’t really going to do anything, but it should be a valid workflow.</p>
